### ISSUE TITLE
Envision is missing packages

### ISSUE THREAD

**Issue by sophronesis** [NixOS/nixpkgs#358664]
## Describe the bug

I've tried using [Envision](https://search.nixos.org/packages?channel=24.05&show=envision&from=0&size=50&sort=relevance&type=packages&query=envision) and wasn't able to proceed further due to missing packages warning. Same on unwrapped version, but with more missing packages

## Steps To Reproduce

1. $ nix-shell -p envision --run envision
2. Press "Build profile"
 
## Expected behavior

Generation of VR profile for Valve Index

## Screenshots

![image](https://github.com/user-attachments/assets/1d0dd7ce-4d5b-4105-a064-3859bb1345e4)

## Additional context

Hyprland + Framework 16 (tried both expansion card slots and direct connection to GPU)

## Metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.6.62, NixOS, 25.05 (Warbler), 25.05.20241119.23e89b7`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.10`
 - nixpkgs: `/nix/store/c9wv7i0af6mysmy65x6nvyfw5izzxv4g-source`

## Notify maintainers

@Pandapip1, @Scrumplex

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by urioTV** [comment]
Right now there is an issue with missing libnotify dependency. However it does not appear in pop-up window but in failed build logs. 
That problem also appears in other distros but there you just install libnotify. In Nix land it should be added to package.

**Comment by Pandapip1** [comment]
I'd like to keep this issue open for tracking when envision is missing a dependency, and to link to it in the envision missing dependencies alert. Thoughts?

**Comment by txkyel** [comment]
> Right now there is an issue with missing libnotify dependency

Is there a way I could patch this for the time being without having to wait on #372276 to become available on unstable?

**Comment by Pandapip1** [comment]
> Is there a way I could patch this for the time being without having to wait on https://github.com/NixOS/nixpkgs/pull/372276 to become available on unstable?

Are you using flakes or channels? If you're using flakes, you can just add my branch as an additional input.

CC @Scrumplex, would really like that merged fast thanks!

**Comment by txkyel** [comment]
The changes in #372276 should be on unstable now according to https://nixpk.gs/pr-tracker.html?pr=372276 but I'm still encountering the same libnotify issue when attempting to build wivrn.

<details>

<summary>envision build log</summary>

```
Building WiVRn...
Already on 'release'
Your branch is up to date with 'origin/release'.
Already up to date.
-- The C compiler identification is GNU 14.2.1
-- The CXX compiler identification is GNU 14.2.1
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.29.2")
-- Found Vulkan: /usr/lib/libvulkan.so (found version "1.3.296") found components: glslc glslangValidator
-- Found Git: /usr/bin/git (found version "2.47.0")
-- Found OpenSSL: /usr/lib/libcrypto.so (found version "3.3.2")
CMake Warning (dev) at CMakeLists.txt:83 (find_package):
  Policy CMP0167 is not set: The FindBoost module is removed.  Run "cmake
  --help-policy CMP0167" for policy details.  Use the cmake_policy command to
  set the policy and suppress this warning.

This warning is for project developers.  Use -Wno-dev to suppress it.

-- Found Boost: /usr/lib64/cmake/Boost-1.86.0/BoostConfig.cmake (found suitable version "1.86.0", minimum required is "1.75.0") found components: locale
-- Checking for modules 'libavcodec;libavutil;libswscale;libavfilter'
--   Found libavcodec, version 61.19.100
--   Found libavutil, version 59.39.100
--   Found libswscale, version 8.3.100
--   Found libavfilter, version 10.4.100
-- Checking for module 'libdrm'
--   Found libdrm, version 2.4.123
-- Checking for module 'x264'
--   Found x264, version 0.164.x
-- Checking for module 'libsystemd'
--   Found libsystemd, version 256
-- Checking for module 'libpipewire-0.3'
--   Found libpipewire-0.3, version 1.2.7
-- Checking for module 'libpulse'
--   Found libpulse, version 17.0
-- Checking for modules 'avahi-client;avahi-glib'
--   Found avahi-client, version 0.8
--   Found avahi-glib, version 0.8
Package sysprof-capture-4 was not found in the pkg-config search path.
Perhaps you should add the directory containing `sysprof-capture-4.pc'
to the PKG_CONFIG_PATH environment variable
Package 'sysprof-capture-4', required by 'glib-2.0', not found
Package sysprof-capture-4 was not found in the pkg-config search path.
Perhaps you should add the directory containing `sysprof-capture-4.pc'
to the PKG_CONFIG_PATH environment variable
Package 'sysprof-capture-4', required by 'glib-2.0', not found
Package sysprof-capture-4 was not found in the pkg-config search path.
Perhaps you should add the directory containing `sysprof-capture-4.pc'
to the PKG_CONFIG_PATH environment variable
Package 'sysprof-capture-4', required by 'glib-2.0', not found
Package sysprof-capture-4 was not found in the pkg-config search path.
Perhaps you should add the directory containing `sysprof-capture-4.pc'
to the PKG_CONFIG_PATH environment variable
Package 'sysprof-capture-4', required by 'glib-2.0', not found
-- Found nlohmann_json: /usr/share/cmake/nlohmann_json/nlohmann_jsonConfig.cmake (found version "3.11.3")
-- Checking for modules 'glib-2.0;gio-2.0;gio-unix-2.0'
--   Found glib-2.0, version 2.82.1
--   Found gio-2.0, version 2.82.1
--   Found gio-unix-2.0, version 2.82.1
Package sysprof-capture-4 was not found in the pkg-config search path.
Perhaps you should add the directory containing `sysprof-capture-4.pc'
to the PKG_CONFIG_PATH environment variable
Package 'sysprof-capture-4', required by 'glib-2.0', not found
Package sysprof-capture-4 was not found in the pkg-config search path.
Perhaps you should add the directory containing `sysprof-capture-4.pc'
to the PKG_CONFIG_PATH environment variable
Package 'sysprof-capture-4', required by 'glib-2.0', not found
Package sysprof-capture-4 was not found in the pkg-config search path.
Perhaps you should add the directory containing `sysprof-capture-4.pc'
to the PKG_CONFIG_PATH environment variable
Package 'sysprof-capture-4', required by 'glib-2.0', not found
Package sysprof-capture-4 was not found in the pkg-config search path.
Perhaps you should add the directory containing `sysprof-capture-4.pc'
to the PKG_CONFIG_PATH environment variable
Package 'sysprof-capture-4', required by 'glib-2.0', not found
-- Checking for module 'libnotify'
--   Package 'gdk-pixbuf-2.0', required by 'libnotify', not found
CMake Error at /nix/store/38cffsqqx823crf1i4bcf6zz1qz1hgpd-cmake-3.30.5/share/cmake-3.30/Modules/FindPkgConfig.cmake:645 (message):
  The following required packages were not found:

   - libnotify

Call Stack (most recent call first):
  /nix/store/38cffsqqx823crf1i4bcf6zz1qz1hgpd-cmake-3.30.5/share/cmake-3.30/Modules/FindPkgConfig.cmake:873 (_pkg_check_modules_internal)
  CMakeLists.txt:179 (pkg_check_modules)


-- Configuring incomplete, errors occurred!

```

</details>

**Comment by txkyel** [comment]
I've added your branch for #374373 and now WiVRn builds correctly, but when attempting to start I get the following error:

```
/home/kyle/.local/share/envision/prefixes/wivrn_default/bin/wivrn-server: error while loading shared libraries: libavcodec.so.61: cannot open shared object file: No such file or directory
WiVRn exited with code 127
```

**Comment by Pandapip1** [comment]
Oh for goodness sake. How the heck does that even happen?! Time for PR number three.

**Comment by Pandapip1** [comment]
So a lot of these dependencies should have been added with:

```
      pkgs.wivrn.buildInputs
      ++ pkgs.wivrn.nativeBuildInputs
```

I don't know why they haven't been, and I need to investigate that. For the time being, I'll manually add all the missing deps. Again.

**Comment by ToasterUwU** [comment]
Im also running into lots of annoying issues.

Currently i cant build the default lighthouse profile since it complains about mesa-common-dev missing, i can avoid that popup with "--skip-dependency-check", but even then i run into lots of issues.

I can now build the profile, but when using the steamvr calibrate option, it tries to run lsusb, which is not part of the dependencies of the package, so unless i put usbutils in a nix-shell with it, it wont be able to even do the calibration. 
And with that workaround in place as well, i can try to start the actual software for VR (Monado), but that fails with "error while loading shared libraries: librealsense2.so.2.56: cannot open shared object file: No such file or directory"

Happy to make a seperate issue for this if it would be more appropriate. All i know is that envision for my setup isnt usable at all atm, and that waiting for a cli interface to be made will take a long time. SteamVR is also just broken for me once again, so i was hoping to use monado via envision as a plan b, so rn there is no way to use VR.
If the issue is that you dont want to manually test for all cases and see if everything is installed all the time, im happy to take that tedious process (at least for the lighthouse profile) into my own hands until automatic tests are possible.

**Comment by Pandapip1** [comment]
Oh shoot I totally forgot about this. I'll investigate it this evening. 

I'm tempted to try to find a way to integrate the nix-index DB. It just continues to have endless errors.

**Comment by Pandapip1** [comment]
Whoops, so much for 'this evening'. I had an onslaught of psets that seem to be consuming pretty much all the free time I had, and likely will have for the next couple days. If anyone wants to debug this, the relevant files are at https://gitlab.com/gabmus/envision/-/tree/main/src/depcheck. grep for the missing dependencies, and add the necessary packages/outputs to https://github.com/NixOS/nixpkgs/blob/master/pkgs/by-name/en/envision/package.nix.
